### PR TITLE
teach IBD to produce KeyTables

### PR DESCRIPTION
**Pull request by danking**

### CODE REVIEW

File: python/hail/dataset.py [hunk 1]
```diff
@@ -2003,24 +2003,36 @@ def hardcalls(self):
...
+        The implementation is based on the IBD algorithm described in the `PLINK
+        paper <http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1950838>`_.
 
-        :py:meth:`~hail.VariantDataset.ibd` requires the dataset to be bi-allelic (otherwise run :py:meth:`~hail.VariantDataset.split_multi`) and does not perform LD pruning. Linkage disequilibrium may bias the result so consider filtering variants first.
+        :py:meth:`~hail.VariantDataset.ibd` requires the dataset to be
+        bi-allelic (otherwise run :py:meth:`~hail.VariantDataset.split_multi`)
+        and does not perform LD pruning. Linkage disequilibrium may bias the
+        result so consider filtering variants first.
+
+        The resulting :py:class:`.KeyTable` keys are pairs of samples with type
```
Comment (cseed) [inline python/hail/dataset.py]: Put the KeyTable signature in a code block.  Keys + values is the wrong abstraction, give the full key table schema and then the (ordered) list of keys.

File: python/hail/dataset.py [hunk 2]
```diff
@@ -2003,24 +2003,36 @@ def hardcalls(self):
...
+        *{ i: String, j: String }*. The values have type: *{ ibd: { Z0: Double,
+        Z1: Double, Z2: Double, PI_HAT: Double }, ibs0: Long, ibs1: Long, ibs2:
+        Long }*
 
         Conceptually, the output is a symmetric, sample-by-sample matrix. The output .tsv has the following form
```
Comment (cseed) [inline python/hail/dataset.py]: Prefer TSV to .tsv.

**Comment by danking** [comment]
@cseed addressed.

**Comment by danking** [comment]
@cseed rebased

File: python/hail/dataset.py [hunk 3]
```diff
@@ -1946,36 +1946,47 @@ def hardcalls(self):
...
 
-        Conceptually, the output is a symmetric, sample-by-sample matrix. The output .tsv has the following form
+        The resulting :py:class:`.KeyTable` entries have the type: *{ i: String,
+        j: String, ibd: { Z0: Double, Z1: Double, Z2: Double, PI_HAT: Double },
+        ibs0: Long, ibs1: Long, ibs2: Long }*. The key list is: `*i: String, j:
+        String*`.
 
-        .. code-block: text
+        Conceptually, the output is a symmetric, sample-by-sample matrix. The
+        output Key Table has the following form
```
Comment (cseed) [inline python/hail/dataset.py]: We're just calling them key tables elsewhere (see keytable.py for example).

**Comment by danking** [comment]
@cseed changed "Key Table" to "key table"

**Pull request merged**